tests/test_all_rules.py

```diff
@@ -584,6 +585,26 @@ def test_non_bbr_in_correct_dir(self):
 class TestRuleMetadata(BaseRuleTest):
     """Test the metadata of rules."""
 
+    def test_dates_not_in_future(self):
+        """Ensure creation and updated dates are not in the future."""
+        invalid = []
+        today = datetime.now(timezone.utc).date()
+
+        for rule in self.all_rules:
+            created_str = rule.contents.metadata.creation_date
+            updated_str = rule.contents.metadata.updated_date
+
+            created = datetime.strptime(created_str, "%Y/%m/%d").date()
```

Comment:
I think this is a good approach at the moment. In the future we may want to prefer using an approach using `dateutil` to parse the date as a more robust method.